brave/src/main/java/brave/internal/Lists.java

```diff
@@ -29,19 +29,19 @@
     return mutable;
   }
 
-  public static List<Object> ensureImmutable(List<Object> extra) {
-    if (isImmutable(extra)) return extra;
+  public static List<Object> ensureImmutable(List<Object> list) {
+    if (list.isEmpty()) return Collections.emptyList();
     // Faster to make a copy than check the type to see if it is already a singleton list
-    if (extra.size() == 1) return Collections.singletonList(extra.get(0));
-    return Collections.unmodifiableList(new ArrayList<>(extra));
+    if (list.size() == 1) return Collections.singletonList(list.get(0));
+    if (isImmutable(list)) return list;
+    return Collections.unmodifiableList(new ArrayList<>(list));
   }
 
   static boolean isImmutable(List<Object> extra) {
-    if (extra == Collections.EMPTY_LIST) return true;
+    if (extra == Collections.emptyList()) return true;
     // avoid copying datastructure by trusting certain names.
     String simpleName = extra.getClass().getSimpleName();
-    return simpleName.equals("SingletonList")
```

Comment:
that SingletonList isn't here hurts my brain because it is a well known type. that said I understand why you are doing this.. it is under the assumption nothing else is using this method which is fair enough. It might hurt brain less to say intentionally not checking SingletonList due as the only call site wont' have it

Reply:
Good call, added a comment both for that and because I realized the empty list check was similarly redundant.